src/dbnode/storage/index/block.go

```diff
@@ -768,6 +778,31 @@ func (b *block) executorWithRLock() (search.Executor, error) {
 	return executor.NewExecutor(readers), nil
 }
 
+func (b *block) segmentsWithRLock() ([]segment.Segment, error) {
```

Comment:
nit: doesn't look like this needs to return an error

Reply:
will do